free-3.md

```diff
@@ -9,42 +9,94 @@ good way to meet their needs.
+people new to open source, going into GitHub and making sense of a large project
+can be very intimidating. I imagine this stops a lot of people from even beginning
+to scratch the surface in understanding how an interesting project works. If I
+could help, the experience of jumping into a new code base would be much less
+intimidating. While READMEs do a good job of explaining the project as a whole,
+there is often a lack of description of how the codebase is structured and what
+each directory contains. Therefore, I would want to be able to visualize the 
+code in relation to the application. There would be a "starting point" directory
+that is clearly defined for each part of the application (front-end, database, 
+back-end, etc.). Ideally there would also be a visualization to trace through 
+which part of the code a specific action runs through. 
 
 ### Why a language?
 _Why is a DSL appropriate for your user(s)? How does it address the need?_
-
+It is appropriate because the target users are developers who write code.
+It makes sense to give them additional code to write because the code will be
+written in the existing code base. Furthermore, it requires language to
+communicate what exactly you want users to understand.
 
 ### Why you?
 _What excites you about this idea? How did you come up with it?_
-
+The motivation is from my summer internship.  I worked at (Sourcegraph)[https://sourcegraph.com], a semantic code search and global cross reference engine. One
+of the ideas thrown around was to help users understand code bases. The idea of
+a "starting point", was explicitly brought up by one of the developers, and I
+want to take this to the next level. This excites me because if done well, it
+could truly be used in every single code base and helpful to every developer
+out there.
 
 ### Domain
 _Describe the project's domain in five words._
-
+Visualizing code repositories
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
-
-
+The user would have to explicitly code where the "starting point" and trace
+points are in the repository. So, we could imagine the code in the dsl to look
+like this: 
+
+```
+import Visualizer
+
+var tracingSomeAction = new Visualizer();
+function foo() {
+	tracingSomeAction.startingPoint()
+	...
+}
+
+function bar() {
+	tracingSomeAction.tracePoint()
+}
+
+function baz() {
+	tracingSomeAction.endPoint()	
+}
+```
+
+The output would be a series of charts or trees for each Visualizer object so
+we could see where to start and the trace through the code for the action.
+ 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
-
+When the program runs, an image with links to specific parts of code would be
```

Comment:
I'm a little confused. Does the visualization represent files in the repo or specific functions and where they are used within the repo?